ext/wiki/main.php

```diff
@@ -332,6 +346,78 @@ public static function get_page(string $title, int $revision=-1): WikiPage
     }
 
+    public static function format_tag_wiki_page(WikiPage $page) {
+        global $database, $config;
+        
+        $row = $database->get_row("
+                SELECT *
+                FROM tags
+                WHERE tag = :title
+                    ", ["title"=>$page->title]);
+        
+        if (!empty($row)) {
+            $template = $config->get_string(WikiConfig::TAG_PAGE_TEMPLATE);
+            
+            //CATEGORIES
+            if (class_exists("TagCategories")) {
+                $tagcategories = new TagCategories;
+                $tag_category_dict = $tagcategories->getKeyedDict();
+            }
+            
+            //ALIASES
+            $aliases = $database->get_col("
+                SELECT oldtag
+                FROM aliases
+                WHERE newtag = :title
+                ORDER BY oldtag ASC
+                    ", ["title"=>$row["tag"]]);
+            
+            if (!empty($aliases)) {
+                $template = str_replace("{aliases}", implode(", ", $aliases), $template);
+            } else {
+                $template = str_replace("{aliases}", $config->get_string(WikiConfig::EMPTY_TAGINFO), $template);
+            }
+            
+            //Things before this line will be passed through html_escape.
+            $template = format_text($template);
+            //Things after this line will NOT be escaped!!! Be careful what you add.
+            
+            if (class_exists("AutoTagger")) {
+                $auto_tags = $database->get_one("
+                    SELECT additional_tags
+                    FROM auto_tag
+                    WHERE tag = :title
+                        ", ["title"=>$row["tag"]]);
+                
+                if (!empty($auto_tags)) {
+                    $auto_tags = Tag::explode($auto_tags);
+                    $f_auto_tags = [];
+                    
+                    $tag_list_t = new TagListTheme;
+                    
+                    foreach ($auto_tags as $a_tag) {
+                        $a_row = $database->get_row("
+                            SELECT *
+                            FROM tags
+                            WHERE tag = :title
+                                ", ["title"=>$a_tag]);
+                        
+                        $tag_html = $tag_list_t->return_tag($a_row, $tag_category_dict ?? []);
+                        array_push($f_auto_tags, $tag_html[1]);
+                    }
+                    
+                    $template = str_replace("{autotags}", implode(", ", $f_auto_tags), $template);
+                } else {
+                    $template = str_replace("{autotags}", format_text($config->get_string(WikiConfig::EMPTY_TAGINFO)), $template);
+                }
+            }
+        }
+        
+        //Insert page body AT LAST to avoid replacing its contents with the actions above.
+        $formatted = str_replace("{body}", format_text($page->body), $template ?? "{body}");
```

Comment:
This looks like it gets applied to all wiki pages, not just tag pages - so even "Site FAQ" etc will be put into the "tag page" template?

Reply:
I coded this months ago but I'm pretty sure I did it with that in mind. In fact, that is the only line that is outside of the "if tag exists" block.
IIRC I did this because the function gets called as a replacement of [what was there](https://github.com/shish/shimmie2/pull/808/commits/a17e2eca15c7187c96885c7b335701006d1edcfa#diff-478b29418b9aaaf7ae189941a14fbc1451c388e1b4ae97d52e2a3f434a404992L80-R80).

If you notice, the last parameter in the line you cited is actually checking for the template, and if it is not assigned, it assumes a template "{body}" which means you will only get the body as without this change.
I'm not saying this is great design, but it works. Also, the idea of replacing the body after everything else is to avoid {thingslikethis} written in the wiki page itself to be replaced hehe

If you prefer a different method of deciding what to format and what to keep tho, I can try to make the changes.